federatedscope/core/auxiliaries/data_builder.py

```diff
@@ -572,14 +572,20 @@ def get_data(config):
         data, modified_config = load_mf_dataset(config)
     elif '@' in config.data.type.lower():
         data, modified_config = load_external_data(config)
+    elif config.data.type is None or config.data.type == "":
+        # The participant (only for server in this version) does not own data
+        data = None
+        modified_config = config
     else:
         raise ValueError('Data {} not found.'.format(config.data.type))
 
     if config.federate.mode.lower() == 'standalone':
         return data, modified_config
     else:
         # Invalid data_idx
-        if config.distribute.data_idx not in data.keys():
+        if config.distribute.data_idx == -1:
```

Comment:
When `data.type` is `None` or `""` while `distribute.data_idx` is given, we should assert in  `configs`.

Reply:
When `date.type` is None or "", the `distribute.data_idx` does not work. IMO, we can provide a WARNING but not an assertion here.